src/nss_wrap.cc

```diff
@@ -0,0 +1,560 @@
+#include "nss_wrap.h"
+
+namespace node {
+namespace nss_wrap {
+
+using nss_module::nss_gethostbyname3_r;
+using nss_module::nss_gethostbyname4_r;
+using nss_module::nss_gethostbyaddr2_r;
+using v8::Array;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Handle;
+using v8::HandleScope;
+using v8::Integer;
+using v8::Local;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+NSSReqWrap::NSSReqWrap(Environment* env,
+                       Local<Object> req_wrap_obj,
+                       char* host_val,
+                       int8_t host_type_val)
+    : ReqWrap(env, req_wrap_obj, AsyncWrap::PROVIDER_NSSREQWRAP),
```

Comment:
So I'm not exactly sure how I can (easily) test the new provider since its usage only comes into play when it finds a non-`file`/`dns` entry in `/etc/nsswitch.conf`, which can't be assumed for all systems.